geomesa-gt/geomesa-gt-partitioning/src/main/scala/org/locationtech/geomesa/gt/partition/postgis/dialect/PartitionedPostgisDialect.scala

```diff
@@ -490,4 +459,33 @@ object PartitionedPostgisDialect extends Conversions {
     def getCoordinateDimensions: Option[Int] =
       Option(d.getUserData.get(Hints.COORDINATE_DIMENSION)).map(int)
   }
+
+  /**
+   * Get a list of indexed columns for the given SimpleFeatureType
+   *
+   * @param cx connection
+   * @param typeName feature type name
+   * @return a sequence of SimpleFeatureType attribute names which have an index
+   */
+  def getIndexedColumns(cx: Connection, typeName: String): List[String] = {
+    val attributesWithIndicesSql =
+      s"""select distinct(att.attname) as indexed_attribute_name
+         |from pg_class obj
+         |join pg_index idx on idx.indrelid = obj.oid
+         |join pg_attribute att on att.attrelid = obj.oid and att.attnum = any(idx.indkey)
+         |where obj.relname = concat(?, ${PartitionedTableSuffix.quoted})
+         |order by att.attname;""".stripMargin
+    try {
+      WithClose(cx.prepareStatement(attributesWithIndicesSql)) { statement =>
+        statement.setString(1, typeName)
+        WithClose(statement.executeQuery()) { rs =>
+          Iterator.continually(rs).takeWhile(_.next()).map(_.getString(1)).toList
+        }
+      }
+    } catch {
+      case NonFatal(e) =>
```

Comment:
it might be better to not catch this error here, and handle it on the caller side, so that the caller can decide if it wants to continue or not. (in the dialect, we'd just log a warning and continue, but other callers might want to take different action)